### ISSUE TITLE
Deferreds are Beautiful documentation has incorrect output

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3941]
|<img alt="ezyang's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| ezyang reported|
|-|-|
|Trac ID|trac#3941|
|Type|defect|
|Created|2009-07-29 18:15:18Z|
|Branch|https://github.com/twisted/twisted/tree/deferred-ex2-doc-3941-2|

In "Exceptions will only be handled by errbacks", the ostensible output is:

```
callback 1
        got result: success
        about to raise exception
errback
we got an exception: Traceback (most recent call last):
  File "./deferred_ex2.py", line 85, in ?
    nonDeferredExample("success")
--- <exception caught here> ---
  File "./deferred_ex2.py", line 46, in nonDeferredExample
    result = failAtHandlingResult(result)
  File "./deferred_ex2.py", line 35, in failAtHandlingResult
    raise RuntimeError, "whoops! we encountered an error"
exceptions.RuntimeError: whoops! we encountered an error
```

This doesn't actually make much sense, since nonDeferredExample is a non-existent function in the code.

The real output should look something like:

```
callback 1
	got result: success
	about to raise exception
errback
we got an exception: Traceback (most recent call last):
  File "deferred_ex2.py", line 85, in <module>
    behindTheScenes("success")
--- <exception caught here> ---
  File "deferred_ex2.py", line 46, in behindTheScenes
    result = failAtHandlingResult(result)
  File "deferred_ex2.py", line 35, in failAtHandlingResult
    raise RuntimeError, "whoops! we encountered an error"
exceptions.RuntimeError: whoops! we encountered an error


-------------------------------------------------

callback 1
	got result: success
	about to raise exception
errback
we got an exception: Traceback (most recent call last):
  File "deferred_ex2.py", line 88, in <module>
    deferredExample()
  File "deferred_ex2.py", line 81, in deferredExample
    d.callback("success")
  File "/usr/lib/python2.6/dist-packages/twisted/internet/defer.py", line 243, in callback
    self._startRunCallbacks(result)
  File "/usr/lib/python2.6/dist-packages/twisted/internet/defer.py", line 312, in _startRunCallbacks
    self._runCallbacks()
--- <exception caught here> ---
  File "/usr/lib/python2.6/dist-packages/twisted/internet/defer.py", line 328, in _runCallbacks
    self.result = callback(self.result, *args, **kw)
  File "deferred_ex2.py", line 35, in failAtHandlingResult
    raise RuntimeError, "whoops! we encountered an error"
exceptions.RuntimeError: whoops! we encountered an error
```

Although I'm a little confused why the sample wants to run itself twice.

<details><summary>Searchable metadata</summary>

```
trac-id__3941 3941
type__defect defect
reporter__ezyang ezyang
priority__normal normal
milestone__ 
branch__branches_deferred_ex2_doc_3941_2 branches/deferred-ex2-doc-3941-2
branch_author__thijs__exarkun thijs, exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__documentation documentation
time__1248891318000000 1248891318000000
changetime__1302747900000000 1302747900000000
version__None None
owner__exarkun exarkun

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph set owner to @thijstriemstra|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_2"></a>|@thijstriemstra commented|
|-|-|

(In [30549]) Branching to 'deferred-ex2-doc-3941'

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_3"></a>|@thijstriemstra commented|
|-|-|

(In [30550]) Correct example output in deferred documentation, refs [#3941](https://github.com/twisted/twisted/issues/3941)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_4"></a>|@thijstriemstra commented|
|-|-|

(In [30551]) Add news file, refs [#3941](https://github.com/twisted/twisted/issues/3941)

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)|@thijstriemstra removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="jonathanj's avatar" src="https://avatars.githubusercontent.com/u/112166?s=50" width="50" height="50">](https://github.com/jonathanj)<a name="note_6"></a>|@jonathanj set owner to @thijstriemstra|
|-|-|

 * Can you capitalise the "i" in "in the next example":

   ```occurs will be the next method called. (in the next example we'll see what happens in the 'chain' after an errback).```

 * The full path to _your_ Twisted installation in the traceback is a little jarring, although technically correct, considering the previous paragraph says things about the traceback being slightly edited.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_8"></a>|@exarkun commented|
|-|-|

(In [31660]) Branching to 'deferred-ex2-doc-3941-2'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_9"></a>|@exarkun set owner to @exarkun<br>@exarkun set status to `assigned`|
|-|-|

Also worth making explicit is that the document still only contains _half_ of the output that `deferred_ex2.py` will generate.  The ticket description is concerned with the fact that the first half is missing entirely.  I think this is fine though, the first half isn't really about Deferreds, so it would just clutter the page.  Also, this whole document is sort of confusing, and changing this one thing isn't going to make any difference to that overall.

I changed the paths in the traceback to be more general and fixed the deferred_ex2.py module docstring (and some other trivial issues in it).  Merging.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_10"></a>|@exarkun set status to `closed`|
|-|-|

(In [31665]) Merge deferred-ex2-doc-3941-2

Author: thijs
Reviewer: jonathanj, exarkun
Fixes: [#3941](https://github.com/twisted/twisted/issues/3941)

Correct sample output in the "deferred in depth" document.
